mdsuite/file_io/extxyz_files.py

```diff
@@ -0,0 +1,256 @@
+                species_name_to_line_idx_dict=species_dict,
+                property_to_column_idx_dict=property_dict,
+                n_header_lines=self.n_header_lines,
+                n_partilces=n_particles,
+                header_lines_for_each_config=True,
+            )
+
+    def _get_metadata(self):
+        """
+        Gets the metadata for database creation as an implementation of the parent class virtual function.
+        """
+        with open(self.file_path, "r") as file:
+            file.readline()
+            # box_l in second header line
+            header = file.readline()
+            box_l = _get_box_l(header)
+
+            file.seek(0)
+            mdsuite.file_io.tabular_text_files.skip_n_lines(
+                file,
+                self.tabular_text_reader_data.n_partilces + self.n_header_lines + 1,
+            )
+            header_1 = file.readline()
+            sample_rate = int(round(_get_time(header_1) - _get_time(header)))
+
+        species_list = mdsuite.file_io.tabular_text_files.get_species_list_from_tabular_text_reader_data(
+            self.tabular_text_reader_data
+        )
+
+        mdata = mdsuite.database.simulation_database.TrajectoryMetadata(
+            n_configurations=self.tabular_text_reader_data.n_configs,
+            box_l=box_l,
+            sample_rate=sample_rate,
+            species_list=species_list,
+        )
+
+        return mdata
+
+    def _get_species_information(self, file, species_idx: int, n_particles: int):
+        """
+        Get the initial species information
+
+        Parameters
+        ----------
+
+        """
+        mdsuite.file_io.tabular_text_files.skip_n_lines(file, self.n_header_lines)
+        # read one configuration
+        traj_data = np.stack(
+            [list(file.readline().split()) for _ in range(n_particles)]
+        )
+
+        # Loop over atoms in first configuration.
+        species_dict = {}
+        for i, line in enumerate(traj_data):
+            sp_name = line[species_idx]
+            if sp_name not in list(species_dict.keys()):
+                species_dict[sp_name] = []
+            species_dict[sp_name].append(i)
+
+        return species_dict
+
+
+def _get_box_l(header: str) -> list:
+    data = copy.deepcopy(header).split()
+    lattice = None
+    start = None
+    stop = None
+    for idx, item in enumerate(data):
+        if "Lattice" in item:
+            start = idx
+            break
+
+    if start is not None:
+        for idx, item in enumerate(data[start:]):
+            if item[-1] == '"':
+                stop = idx
+                break
+    else:
+        raise RuntimeError("Could not find lattice size in file header")
+
+    if stop is not None:
+        lattice = data[start : stop + 1]
+        lattice[0] = lattice[0].split("=")[1].replace('"', "")
+        lattice[-1] = lattice[-1].replace('"', "")
+        lattice = np.array(lattice).astype(float)
+
+    return [lattice[0], lattice[4], lattice[8]]
```

Comment:
does this return the diagonal elements of the box vectors? maybe add a docstring?